BigWatson/app/logic/helpers.py

```diff
@@ -24,23 +25,34 @@ def parse_discovery_results(self, results):
         Parses results received from querying Watson Discovery.
         Returns list of Articles parsed from json results.
         """
-        
+
+        #async get article data
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(loop)
+        futures = [self.coro_get_article_data(results[i]['url'], i) for i in range(len(results))]
+        loop_result = loop.run_until_complete(asyncio.wait(futures))
+        tasks = loop_result[0]
+        task_results = [task.result() for task in tasks]
+        task_results.sort()
+        loop.close()
+
+        #create article objects with data
         discovery_results = []
-        for result in results:
-            title = result['title']
-            url = result['url']
+        for i in range(len(results)):
```

Comment:
Do `for i, result in enumerate(results):` I think that'll clean up this loop a bit.